app/src/services/subscribers/CensusEventSubscriberService.ts

```diff
@@ -98,16 +98,22 @@ export default class CensusEventSubscriberService implements ServiceInterface {
 
         // Set up event handlers
         this.wsClient.on('death', (event) => {
-            if (this.checkAlert(event)) {
-                const deathEvent = new DeathEvent(event, this.getAlert(event));
+            const worldId = parseInt(event.world_id, 10);
```

Comment:
Kind of liked the helper function, specially as we will need to subscribe to more event with this same logic. Maybe we should create a helper function that creates a function that is passed as a listener instead?

Reply:
Can't do much, base class does not contain the zoneId.

You mean filter function?

Reply:
We are not using the base class anymore as we are listening to the events directly. So `death` event will always be `Death` objects.

Reply:
@marci4 Zone ID is figured out from the injected ActiveAlert, hence why it's injected.

Reply:
@microwavekonijn Would be awesome if you can send zoneID down the pipe though.

Reply:
then I misunderstood you with the function? Could you explain this please?

Reply:
`checkAlert` was there because `getAlert` couldn't accept anything else other than an `MetagameEvent`, if that has now been fixed because Zone is now supplied we can safely get rid of it and just use the `ActiveAlertAuthority`'s `alertExists()`

Reply:
@marci4 every listener will have the if statement, so we need to unduplicate that code or keep the duplication as minimal as possible.

@Maelstromeous down the pipe?

Reply:
@microwavekonijn as part of the death / zone supported events, but not entirely sure you can figure that one out, we can as we have the Alert context which contains the zone and world IDs for all attached events.

Reply:
@microwavekonijn I thought you meant something completely different, my bad

Reply:
Certain events have a `zone_id` but not all. Only login/logout and metagameevent do not have that one.